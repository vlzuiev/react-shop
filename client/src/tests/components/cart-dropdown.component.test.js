import React from 'react';
import { shallow } from 'enzyme'
import CartDropDown from '../../components/cart-dropdown/cart-dropdown.container';
import Header from '../../components/header/header.component';
 
describe('Header component', () => {
    let wrapper;
    let mockSignOutStart;
    let hidden; 
  
    beforeEach(() => {
      mockSignOutStart = jest.fn();
      hidden = true; 

      const mockProps = {
        hidden: hidden,
        currentUser: {
          uid: '123'
        },
        signOutStart: mockSignOutStart
      };
  
      wrapper = shallow(<Header {...mockProps} />);

    });

    test('should render Header component', () => {  
      expect(wrapper).toMatchSnapshot();
    }); 

    describe('if currentUser is present', () => {
      test('should call signOutStart method when link is clicked', () => { 
        wrapper
        .find('Styled(NavLink)')
        .at(3)
        .simulate('click');
  
        expect(mockSignOutStart).toHaveBeenCalled();
      }); 

      test('should not render CartDropDown', () => {  
        expect(wrapper.containsMatchingElement(<CartDropDown />)).toEqual(false);  
      })

    });

    describe('if currentUser is null', () => {
      test('should render CartDropdown', () => { 
        const mockProps = {
          hidden: false,
          currentUser: null,
          signOutStart: mockSignOutStart
        };
  
        const newWrapper = shallow(<Header {...mockProps} />); 
        expect(newWrapper.containsMatchingElement(<CartDropDown />)).toEqual(true); 

      });
    });
    
});
