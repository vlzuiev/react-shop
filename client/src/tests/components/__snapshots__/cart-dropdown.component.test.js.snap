// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
<styled.div>
  <Styled(NavLink)
    to="/"
  >
    <ForwardRef(SvgCrown)
      className="logo"
    />
  </Styled(NavLink)>
  <styled.div>
    <Styled(NavLink)
      activeClassName="active"
      to="/shop"
    >
      SHOP
    </Styled(NavLink)>
    <Styled(NavLink)
      activeClassName="active"
      to="/contact"
    >
      CONTACT
    </Styled(NavLink)>
    <Styled(NavLink)
      activeClassName="active"
      as="div"
      onClick={[MockFunction]}
      to=""
    >
       SIGN OUT 
    </Styled(NavLink)>
    <Connect(withRouter(FavoriteIcon)) />
    <Connect(CartIcon) />
  </styled.div>
</styled.div>
`;
